### PR TITLE
[polish] Improve Sinks javadoc in anticipation of deprecated emit API removal

### PR DESCRIPTION
**Pull request by simonbasle**

### CODE REVIEW

File: reactor-core/src/main/java/reactor/core/publisher/Sinks.java [hunk 1]
```diff
@@ -191,9 +198,13 @@ public Emission getReason() {
 	}
 
 	/**
-	 * A handler for any non-successful emission result from operations
-	 * such as {@link Many#emitNext(Object, EmitFailureHandler)}
-	 * that allows retrying or failing the operation.
+	 * A handler supporting the emit API (eg. {@link Many#emitNext(Object, EmitFailureHandler)}),
+	 * checking non-successful emission results from underlying {@link Many#tryEmitNext(Object) tryEmit}
+	 * API calls to decide whether or not such calls should be retried.
+	 * Other than instructing to retry, the handler are allowed to have side effects
```
Comment (bsideup) [inline reactor-core/src/main/java/reactor/core/publisher/Sinks.java:204]:
```suggestion
	 * Other than instructing to retry, the handlers are allowed to have side effects
```

File: reactor-core/src/main/java/reactor/core/publisher/Sinks.java [hunk 2]
```diff
@@ -191,9 +198,13 @@ public Emission getReason() {
...
+	 * like parking the current thread for longer retry loops. They don't however influence the
```
Comment (bsideup) [inline reactor-core/src/main/java/reactor/core/publisher/Sinks.java:205]: not 100% sure about this one, but:

```suggestion
	 * like parking the current thread for longer retry loops. They don't, however, influence the
```

**Review by bsideup — DISMISSED** [review]

**Comment by simonbasle** [comment]
@bsideup I also edited the reference documentation (and improved the `GuideTest.java` to keep it in sync). Please re-review.

File: docs/asciidoc/advancedFeatures.adoc [hunk 3]
```diff
@@ -193,21 +193,21 @@ Compare the first example to the second example, shown in the following code:
...
 Flux<String> hotFlux = hotSource.asFlux().map(String::toUpperCase);
 
-
 hotFlux.subscribe(d -> System.out.println("Subscriber 1 to Hot Source: "+d));
 
-hotSource.next("blue")
-         .next("green");
+hotSource.emitNext("blue", FAIL_FAST);
+//note: orThrow is an alternative to emitNext+Sinks.EmitFailureHandler.FAIL_FAST above, suitable for tests
+hotSource.tryEmitNext("green").orThrow();
```
Comment (aneveu) [inline docs/asciidoc/advancedFeatures.adoc:204]: I would have inverted those two lines as `orThrow` is used after the comment and though one may not really understand where does it comes from

**Review by aneveu — APPROVED** [review]

**Review by bsideup — APPROVED** [review]

File: docs/asciidoc/advancedFeatures.adoc (hunk 3 above)
Comment (simonbasle) [reply docs/asciidoc/advancedFeatures.adoc]: I updated to use a callout. found a bad callout in an unrelated asciidoc file and fixed it as a separate commit, so I rebased the PR into 2 commits FYI

**Review by aneveu — APPROVED** [review]

**Pull request merged**